metadatastore/commands.py

```diff
@@ -39,167 +99,512 @@ def inner(*args, **kwargs):
+
+
+def db_disconnect():
+    """Helper function to deal with stateful connections to mongoengine"""
+    mongoengine.connection.disconnect(ALIAS)
+    for collection in [RunStart, RunStop, EventDescriptor,
+                       Event, DataKey]:
+        collection._collection = None
+
+
+def db_connect(database, host, port):
+    """Helper function to deal with stateful connections to mongoengine
+
+    .. warning
+
+       This will silently ignore input if the database is already
+       connected, even if the input database, host, or port are
+       different than currently connected.  To change the database
+       connection you must call `db_disconnect` before attempting to
+       re-connect.
     """
-    return {key: [data_dict['value'], data_dict['timestamp']]
-            for key, data_dict in six.iteritems(event_dict)}
+    return connect(db=database, host=host, port=port, alias=ALIAS)
 
 
 # database INSERTION ###################################################
 
 @_ensure_connection
-def insert_run_start(time, scan_id, beamline_id, beamline_config, uid=None,
-                     owner=None, group=None, project=None, custom=None):
-    """Provide a head for a sequence of events. Entry point for an
-    experiment's run.
+def insert_run_start(time, scan_id, beamline_id, uid,
+                     owner='', group='', project='', custom=None):
+    """Insert a RunStart document into the database.
 
     Parameters
     ----------
     time : float
-        The date/time as found at the client side when an event is
-        created.
+        The date/time as found at the client side when the run is started
     scan_id : int
-        Unique scan identifier visible to the user and data analysis
+        Scan identifier visible to the user and data analysis.  This is not
+        a unique identifier.
     beamline_id : str
-        Beamline String identifier. Not unique, just an indicator of
-        beamline code for multiple beamline systems
-    beamline_config : metadatastore.documents.Document or str
-        if Document:
-            The metadatastore beamline config document
-        if str:
-            uid of beamline config corresponding to a given run
-    uid : str, optional
-        Globally unique id string provided to metadatastore
+        Beamline String identifier.
+    uid : str
+        Globally unique id to identify this RunStart
     owner : str, optional
-        A username associated with the entry
+        A username associated with the RunStart
     group : str, optional
-        A group (e.g., UNIX group) associated with the entry
+        An experimental group associated with the RunStart
     project : str, optional
         Any project name to help users locate the data
     custom: dict, optional
         Any additional information that data acquisition code/user wants
-        to append to the Header at the start of the run.
+        to append to the RunStart at the start of the run.
```

Comment:
Can you note that this `custom` dict is going to be unpacked into the top level of the RunStart document?